File: code_julia/equilibria/axial_z.jl
#=============================================================================
Author: Rayla Kurosaki
GitHub: https://github.com/rkp1503
=============================================================================#

module axial_z

function get_equilibria(parameter_values)
    equilibria = get_analytical_solutions(parameter_values)
    r₂₁, r₃₁, p, ϕ₁₂, ϕ₂₁, ϕ₁₃, u₁, u₂, u₃, u₄ = parameter_values
    for (i, equilibrium) in enumerate(equilibria)
        x, y, z = equilibrium
        # if !(z > 0)
        #     removeat!(equilibria, i)
        # end
        cond = r₃₁ > u₄
        if !all([cond])
            deleteat!(equilibria, i)
        end
    end
    return equilibria
end;

function get_stable_equilibria!(equilibria, parameter_values)
    r₂₁, r₃₁, p, ϕ₁₂, ϕ₂₁, ϕ₁₃, u₁, u₂, u₃, u₄ = parameter_values
    for (i, equilibrium) in enumerate(equilibria)
        x, y, z = equilibrium
        cond1 = (u₄/r₃₁)+(1/ϕ₁₃) < 1
        cond2 = (u₄/r₃₁)+((r₂₁*u₂)/(u₁*(1-p))) < 1
        cond3 = (u₄/r₃₁)<(1/2)
        if !all([cond1, cond2, cond3])
            deleteat!(equilibria, i)
        end
    end
    return nothing
end;

function get_analytical_solutions(parameter_values)
    r₂₁, r₃₁, p, ϕ₁₂, ϕ₂₁, ϕ₁₃, u₁, u₂, u₃, u₄ = parameter_values
    z = 1-(u₄/r₃₁)
    return [(0, 0, z)]
end;
    
end

File: code_python/equilibria/axial_z.py
"""
Author: Rayla Kurosaki

GitHub: https://github.com/rkp1503
"""


def get_equilibria(parameter_values):
    equilibria = get_analytical_solutions(parameter_values)
    r_yx, r_zx, p, phi_xy, phi_yx, phi_xz, u_1, u_2, u_3, u_4 = parameter_values
    for (i, equilibrium) in enumerate(equilibria):
        x, y, z = equilibrium
        # if not (z > 0):
        #     del equilibria[i]
        #     pass
        cond = r_zx > u_4
        if not all([cond]):
            del equilibria[i]
            pass
        pass
    return equilibria


def get_analytical_solutions(parameter_values):
    r_yx, r_zx, p, phi_xy, phi_yx, phi_xz, u_1, u_2, u_3, u_4 = parameter_values
    z = 1 - (u_4 / r_zx)
    return [(0, 0, z)]


def get_stable_equilibria(equilibria, parameter_values):
    r_yx, r_zx, p, phi_xy, phi_yx, phi_xz, u_1, u_2, u_3, u_4 = parameter_values
    for (i, equilibrium) in enumerate(equilibria):
        x, y, z = equilibrium
        cond1 = (u_4 / r_zx) + (1 / phi_xz) < 1
        cond2 = (u_4 / r_zx) + ((r_yx * u_2) / (u_1 * (1 - p))) < 1
        cond3 = (u_4 / r_zx) < (1 / 2)
        if not all([cond1, cond2, cond3]):
            del equilibria[i]
            pass
        pass
    return equilibria
